File: backend/app/models/preferences.py
"""Allowed guest preference values (PRD Section 6.1)."""

from __future__ import annotations

DIETARY_OPTIONS: tuple[str, ...] = (
    "vegetarian",
    "vegan",
    "halal",
    "kosher",
    "gluten_free",
    "nut_allergy",
)

INTEREST_OPTIONS: tuple[str, ...] = (
    "culture",
    "food",
    "outdoor",
    "nightlife",
    "sustainability",
    "wellness",
)

PET_SERVICE_CATEGORY_OPTIONS: tuple[str, ...] = (
    "dog_walker",
    "mobile_grooming",
)

# Map hotel seed `city` to restaurants.json keys
CITY_RESTAURANT_KEYS: dict[str, str] = {
    "New York": "new_york",
    "Washington": "washington_dc",
    "Chicago": "chicago",
    "Miami": "miami",
}

File: frontend/lib/preferences.ts
/** Guest preference options — keep in sync with backend/app/models/preferences.py */

export const DIETARY_OPTIONS = [
  { id: "vegetarian", label: "Vegetarian" },
  { id: "vegan", label: "Vegan" },
  { id: "halal", label: "Halal" },
  { id: "kosher", label: "Kosher" },
  { id: "gluten_free", label: "Gluten-free" },
  { id: "nut_allergy", label: "Nut allergy" },
] as const;

export const INTEREST_OPTIONS = [
  { id: "culture", label: "Culture" },
  { id: "food", label: "Food" },
  { id: "outdoor", label: "Outdoor" },
  { id: "nightlife", label: "Nightlife" },
  { id: "sustainability", label: "Sustainability" },
  { id: "wellness", label: "Wellness" },
] as const;

export const PET_SERVICE_CATEGORY_OPTIONS = [
  { id: "dog_walker", label: "Dog walking & sitting" },
  { id: "mobile_grooming", label: "Mobile grooming" },
] as const;

export type DietaryId = (typeof DIETARY_OPTIONS)[number]["id"];
export type InterestId = (typeof INTEREST_OPTIONS)[number]["id"];
export type PetServiceCategoryId = (typeof PET_SERVICE_CATEGORY_OPTIONS)[number]["id"];
